src/main/java/cz/cvut/kbss/termit/websocket/ResultWithHeaders.java

```diff
@@ -0,0 +1,59 @@
+package cz.cvut.kbss.termit.websocket;
+
+import cz.cvut.kbss.termit.websocket.handler.WebSocketMessageWithHeadersValueHandler;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.springframework.messaging.handler.annotation.SendTo;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Wrapper carrying a result from WebSocket controller
+ * including the {@link #payload}, {@link #destination} and {@link #headers} for the resulting message.
+ * <p>
+ * Do not combine with other method-return-value handlers (like {@link SendTo @SendTo})
```

Comment:
I made this wrapper to be able to reply with custom headers, which would otherwise require constructing a stomp message in the controller.
The returned object is then handled by `WebSocketMessageWithHeadersValueHandler`.
Every value returned from a controller method can be handled only by a single return value handler.
Annotations like `@SendTo`/`@SendToUser` are handled by separate return value handlers, so only one can be used simultaneously.

Reply:
This is fine by me, just explain in the Javadoc why it should not be combined with other method-return-value handlers:)